modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3467,26 +3467,39 @@ def broadcast_apply_full_axis(
                     elif len(new_columns) == 1 and new_partitions.shape[1] == 1:
                         kw["column_widths"] = [1]
```

Comment:
Can we just this change?

```suggestion
                    elif new_partitions.shape[1] == 1:
                        kw["column_widths"] = [len(new_columns)]
```

Reply:
Same for `row_lengths` a few lines above.

Reply:
This is going to happen in https://github.com/modin-project/modin/pull/7133.

Reply:
**Note:** it (https://github.com/modin-project/modin/pull/7131#discussion_r1541960387) can be done instead of the following changes:
```python
                if (
                    kw["column_widths"] is None
                    and ModinIndex.is_materialized_index(new_columns)
                    and len(new_partitions.shape) > 1
                    and new_partitions.shape[1] == 1
                ):
                    kw["column_widths"] = [len(new_columns)]
```

This way the diff will be smaller.

Reply:
Your suggestion relates to the problem #7133 resolves. We should make changes not by diff size but an issue related.

Reply:
> **Note:** it ([#7131 (comment)](https://github.com/modin-project/modin/pull/7131#discussion_r1541960387)) can be done instead of the following changes:
> 
> ```python
>                 if (
>                     kw["column_widths"] is None
>                     and ModinIndex.is_materialized_index(new_columns)
>                     and len(new_partitions.shape) > 1
>                     and new_partitions.shape[1] == 1
>                 ):
>                     kw["column_widths"] = [len(new_columns)]
> ```
> 
> This way the diff will be smaller.

This block of code is independent of `keep_partitioning` parameter, are you planning to get rid of the code duplication in #7133?

Reply:
I am kind of confused. Your suggestion lies under `if not keep_partitioning` branch. If we have duplication in #7133, we can get rid of it there.